devsetup/scripts/edpm-compute-bmaas.sh

```diff
@@ -0,0 +1,274 @@
+#!/bin/bash
+#
+# Copyright 2023 Red Hat Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+set -ex
+RHEL_IMAGE_URL=${RHEL_IMAGE_URL:-"https://images.rdoproject.org/centos9/master/rdo_trunk/current-tripleo/overcloud-hardened-uefi-full.qcow2"}
+AGENT_IMAGE_URL=${AGENT_IMAGE_URL:-"quay.io/openstack-k8s-operators/openstack-baremetal-operator-agent:latest"}
+DOWNLOADER_IMAGE_URL=${DOWNLOADER_IMAGE_URL:-"quay.io/openstack-k8s-operators/openstack-baremetal-operator-downloader:latest"}
+APACHE_IMAGE_URL=${APACHE_IMAGE_URL:-"registry.redhat.io/rhel8/httpd-24:latest"}
+NETWORK_NAME=${NETWORK_NAME:-"crc-bmaas"}
+NODE_NAME_PREFIX=${NODE_NAME_PREFIX:-"crc-bmaas"}
+
+NODE_INDEX=0
+while IFS= read -r instance; do
+    export uuid_${NODE_INDEX}="${instance% *}"
+    name="${instance#* }"
+    export mac_address_${NODE_INDEX}=$(virsh --connect=qemu:///system domiflist "$name" | grep "${NETWORK_NAME}" | awk '{print $5}')
+    NODE_INDEX=$((NODE_INDEX+1))
+done <<< "$(virsh --connect=qemu:///system list --all --uuid --name | grep "${NODE_NAME_PREFIX}")"
+
+cat <<EOF >bmh_deploy.yaml
+---
+# This is the secret with the BMC credentials (Redfish in this case).
+apiVersion: v1
+kind: Secret
+metadata:
+  name: node-1-bmc-secret
+type: Opaque
+data:
+  username: YWRtaW4=
+  password: cGFzc3dvcmQ=
+
+---
+# This is the secret with the BMC credentials (Redfish in this case).
+apiVersion: v1
+kind: Secret
+metadata:
+  name: node-2-bmc-secret
```

Comment:
We are hardcoding values for 2 nodes, I wonder if we should dynamically create these based on value of `BMAAS_NODE_COUNT` passed

Reply:
yeah, we can optimize the script in followup.